conda/environments/all_cuda-115_arch-x86_64.yaml

```diff
@@ -9,7 +9,7 @@ channels:
 - nvidia
 dependencies:
 - aiobotocore>=2.2.0
-- arrow-cpp=9
+- libarrow=10.0.1
```

Comment:
We had previously made this `9` instead of pinning to a specific patch to allow floating forward as new patch versions are released. The Arrow community has strived to ensure patch releases are ABI compatible.

Any issue with making this `libarrow=10` or if the patch is needed then `libarrow>=10.0.1,<11`?

Reply:
Oh yes, now I remember. We can probably make this `libarrow=10`

cc: @robertmaynard for vis

Reply:
Sounds good to me